src/Workspaces/CSharp/Portable/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -222,7 +225,7 @@ private static CSharpSyntaxContext CreateContextWorker(Workspace workspace, Sema
                 isInImportsDirective: IsLeftSideOfUsingAliasDirective(leftToken, cancellationToken),
                 isLabelContext: syntaxTree.IsLabelContext(position, cancellationToken),
                 isTypeArgumentOfConstraintContext: syntaxTree.IsTypeArgumentOfConstraintClause(position, cancellationToken),
-                isRightOfDotOrArrowOrColonColon: syntaxTree.IsRightOfDotOrArrowOrColonColon(position, cancellationToken),
+                isRightOfDotOrArrowOrColonColon: isAfterFirstDotOfDotDot || syntaxTree.IsRightOfDotOrArrowOrColonColon(position, cancellationToken),
```

Comment:
So, i'm ambivalent about this.

There's a reasonably strong argument to be made that this check should be pushed *into* IsRightOfDotOrArrowOrColonColon.  The IsRightOfDotOrArrowOrColonColon is literally the "should i bring up member completion after the user typed the previous character" function... so i'm about 80% feeling like it is the function that should be updated.

Sorry! :)

Reply:
It involves more changes in other files. Hopefully, i made them properly. Please correct me if I'm wrong.

---
In reply to: [254071119](https://github.com/dotnet/roslyn/pull/33148#discussion_r254071119) [](ancestors = 254071119)